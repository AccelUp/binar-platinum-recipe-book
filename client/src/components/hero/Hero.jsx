import classes from "./Hero.module.css";

function Hero() {
  return (
    <div className={classes.Hero}>
      <div className="absolute  inset-0  sm:bg-transparent "></div>

      <div className="relative mt-2  font-Poppins mx-auto max-w-screen-xl px-4 py-32 sm:px-6 lg:flex lg:h-screen lg:items-center lg:px-8">
        <div className="max-w-xl text-center ltr:sm:text-left rtl:sm:text-right">
          <h1 className="text-5xl text-left font-extrabold sm:text-5xl">
            Unlock the Secrets
            <strong className="block font-extrabold text-primary">
              of Delicious Dishes
            </strong>
          </h1>

          <p className="mt-4 text-left max-w-lg sm:text-xl/relaxed">
            Whisking Up Culinary Magic: Where Flavor, Tradition, and Innovation
            Dance Together in a Symphony of Delightful Dishes
          </p>

          <div className="mt-8 flex flex-wrap gap-4 text-center">
            <a
              href="#"
              className="block w-full rounded bg-primary px-12 py-3 text-sm font-medium text-white shadow hover:bg-yellow-400 focus:outline-none focus:ring active:bg-yellow-600 sm:w-auto"
            >
              Get Started
            </a>

            <a
              href="#"
              className="block w-full rounded bg-whitee px-12 py-3 text-sm font-medium text-blackk shadow sm:w-auto"
            >
              Learn More
            </a>
          </div>
        </div>
      </div>
    </div>
  );
}
export default Hero;
