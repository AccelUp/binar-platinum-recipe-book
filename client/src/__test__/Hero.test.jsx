import React from "react";
import Hero from "../components/hero/Hero";
import renderer from "react-test-renderer";

describe("Hero Component", () => {
  it("should render the Hero component correctly", () => {
    const component = renderer.create(<Hero />);
    const tree = component.toJSON();

    // Assert that the rendered component matches the snapshot
    expect(tree).toMatchSnapshot();
  });
});
