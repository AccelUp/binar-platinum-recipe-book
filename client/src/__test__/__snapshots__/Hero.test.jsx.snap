// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero Component > should render the Hero component correctly 1`] = `
<div
  className="_Hero_ad7129"
>
  <div
    className="absolute  inset-0  sm:bg-transparent "
  />
  <div
    className="relative mt-2  font-Poppins mx-auto max-w-screen-xl px-4 py-32 sm:px-6 lg:flex lg:h-screen lg:items-center lg:px-8"
  >
    <div
      className="max-w-xl text-center ltr:sm:text-left rtl:sm:text-right"
    >
      <h1
        className="text-5xl text-left font-extrabold sm:text-5xl"
      >
        Unlock the Secrets
        <strong
          className="block font-extrabold text-primary"
        >
          of Delicious Dishes
        </strong>
      </h1>
      <p
        className="mt-4 text-left max-w-lg sm:text-xl/relaxed"
      >
        Whisking Up Culinary Magic: Where Flavor, Tradition, and Innovation Dance Together in a Symphony of Delightful Dishes
      </p>
      <div
        className="mt-8 flex flex-wrap gap-4 text-center"
      >
        <a
          className="block w-full rounded bg-primary px-12 py-3 text-sm font-medium text-white shadow hover:bg-yellow-400 focus:outline-none focus:ring active:bg-yellow-600 sm:w-auto"
          href="#"
        >
          Get Started
        </a>
        <a
          className="block w-full rounded bg-whitee px-12 py-3 text-sm font-medium text-blackk shadow sm:w-auto"
          href="#"
        >
          Learn More
        </a>
      </div>
    </div>
  </div>
</div>
`;
